controllers/storagecluster/storageclasses.go

```diff
@@ -97,7 +97,7 @@ func (r *StorageClusterReconciler) createStorageClasses(sccs []StorageClassConfi
 		sc := scc.storageClass
 
 		switch {
-		case strings.Contains(sc.Name, "-ceph-rbd") && !scc.isClusterExternal:
+		case (strings.Contains(sc.Name, "-ceph-rbd") || strings.Contains(sc.Provisioner, "rbd.csi.ceph.com")) && !scc.isClusterExternal:
```

Comment:
These changes don't look related to changing names.

Reply:
I did this change, because if we use custom names for SC then we would never hit the conditions in the switch the case i.e, to wait for the backend to be ready before creating the SC using them

Reply:
instead of `rbd.csi.ceph.com` check for `<storagecluster-namespace>.rbd.csi-ceph.com`